src/Core/ObjectRegistry.php

```diff
@@ -331,8 +331,9 @@ public function reset()
      *
      * @param string $objectName The name of the object to set in the registry.
      * @param object $object instance to store in the registry
-     * @psalm-param TObject $object
      * @return $this
+     * @psalm-param TObject $object
+     * @psalm-suppress MoreSpecificReturnType
```

Comment:
Do we really need both this and the one above the return statement?

Reply:
We do if we want to keep pslam happy :)  My understanding of generics annotation is still WIP. So I am not sure whether our annotation are incorrect or it's a bug in pslam, so suppressing the error is the easiest option right now.